src/ResultInterface.php

```diff
@@ -0,0 +1,48 @@
+<?php
+/**
+ * Zend Framework (http://framework.zend.com/)
+ *
+ * @link      http://github.com/zendframework/zf2 for the canonical source repository
+ * @copyright Copyright (c) 2005-2015 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license   http://framework.zend.com/license/new-bsd New BSD License
+ */
+
+namespace Zend\Validator;
+
+/**
+ * This interface provide methods for to know the result of a validation and the reasons when the result is not valid
+ *
+ * This interface resolve the following questions:
+ *    - Is the result valid or invalid?
+ *    - If not, Why is not valid?
+ */
+interface ResultInterface
+{
+    /**
+     * Is the validation result valid?
+     *
+     * @return bool
+     */
+    public function isValid();
+
+    /**
+     * Is the validation result invalid?
+     *
+     * @return bool
+     */
+    public function isNotValid();
```

Comment:
We all know if and while are the most basic statements for control and loops but we all like for() and even more foreach()

Also this may help for all those people wants a whitespace after the not (!) operator.